src/main/java/dev/cdevents/CDEventsGenerator.java

```diff
@@ -0,0 +1,129 @@
+package dev.cdevents;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.mustachejava.DefaultMustacheFactory;
+import com.github.mustachejava.Mustache;
+import com.github.mustachejava.MustacheFactory;
+import dev.cdevents.constants.CDEventConstants;
+import dev.cdevents.exception.CDEventsException;
+import dev.cdevents.models.SchemaData;
+import org.apache.commons.lang.StringUtils;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+public class CDEventsGenerator {
+    private static ObjectMapper objectMapper = new ObjectMapper();
+
+    public static void main(String[] args) {
+        File folder = new File(CDEventConstants.SCHEMA_FOLDER);
+        if (folder.isDirectory()) {
+            File[] files = folder.listFiles((dir, name) -> name.toLowerCase().endsWith(".json"));
+            if (files != null) {
+                //Create Mustache factory and compile event-template.mustache template
+                MustacheFactory mf = new DefaultMustacheFactory();
+                Mustache mustache = mf.compile(CDEventConstants.EVENT_TEMPLATE_MUSTACHE);
+
+                //Generate a class file for each Json schema file using a mustache template
+                for (File file : files) {
+                    SchemaData schemaData = buildCDEventDataFromJsonSchema(file);
+                    generateClassFileFromSchemaData(mustache, schemaData);
+                }
+            }
+        }
+    }
+
+    private static void generateClassFileFromSchemaData(Mustache mustache, SchemaData schemaData) {
+        String classFileName = StringUtils.join(new String[]{schemaData.getCapitalizedSubject(), schemaData.getCapitalizedPredicate(), "CDEvent", ".java"});
+        File classFile = new File("src/main/java/dev/cdevents/events/generated", classFileName);
+        try {
+            FileWriter fileWriter = new FileWriter(classFile);
+            BufferedWriter bufferedWriter = new BufferedWriter(fileWriter);
+            mustache.execute(bufferedWriter, schemaData).flush();
+            fileWriter.close();
+        } catch (IOException e) {
+            throw new CDEventsException("Exception occurred while generating class file from Json schema ", e);
+        }
+        System.out.println("Rendered event-template has been written to file - "+classFile.getAbsolutePath());
```

Comment:
Should this be logged with a logger or printing is fine?

Reply:
yes, will be changing to logger for all the required logs